src/micromamba/config.cpp

```diff
@@ -194,194 +178,150 @@ set_config_path_command(CLI::App* subcom)
+        ->excludes(system_flag)
+        ->excludes(env_flag);
 }
 
+
+enum class SequenceAddType
+{
+    kAppend = 0,
+    kPrepend = 1
+};
+
 void
-set_config_prepend_command(CLI::App* subcom)
+set_config_sequence_command(CLI::App* subcom)
 {
     set_config_path_command(subcom);
 
     auto& config = Configuration::instance();
-    auto& ctx = Context::instance();
-
-    auto& prepend_map = config.insert(
-        Configurable("prepend_map", std::vector<std::string>({ "" }))
-            .group("Output, Prompt and Flow Control")
-            .description("Add one configuration value to the beginning of a list key"));
-    subcom->add_option(
-        "prepend_map", prepend_map.set_cli_config({ "" }), prepend_map.description());
-
-    auto& file_path = config.at("config_set_file_path").get_wrapped<fs::path>();
-    auto& env_path = config.at("config_set_env_path").get_wrapped<bool>();
-    auto& system_path = config.at("config_set_system_path").get_wrapped<bool>();
-
-    subcom->callback([&]() {
-        config.at("use_target_prefix_fallback").set_value(true);
-        config.at("show_banner").set_value(false);
-        config.at("target_prefix_checks")
-            .set_value(MAMBA_ALLOW_EXISTING_PREFIX | MAMBA_ALLOW_MISSING_PREFIX
-                       | MAMBA_ALLOW_NOT_ENV_PREFIX | MAMBA_NOT_EXPECT_EXISTING_PREFIX);
-        config.load();
-
-        fs::path rc_source = env::expand_user(env::home_directory() / ".condarc");
-        std::ofstream rc_file;
-
-        std::string prepend_key = prepend_map.value().front();
-        std::string prepend_value = prepend_map.value().back();
+    auto& specs = config.insert(Configurable("config_set_sequence_spec",
+                                             std::vector<std::pair<std::string, std::string>>({}))
+                                    .group("Output, Prompt and Flow Control")
+                                    .description("Add value to the beginning of a configuration"),
+                                true);
+    subcom->add_option("specs", specs.set_cli_config({}), specs.description())->required();
+}
 
-        if (file_path.configured())
-        {
-            path::touch(file_path.value().string(), true);
-            rc_source = env::expand_user(file_path.value()).string();
-        }
-        else if (env_path.configured())
-        {
-            std::string path = fs::path(ctx.target_prefix / ".condarc");
-            env_path_setup(path, rc_source);
-        }
-        else if (system_path.configured())
-        {
-            system_path_setup(rc_source);
-        }
+void
+set_sequence_to_yaml(YAML::Node& node,
+                     const std::string& key,
+                     const std::string& value,
+                     const SequenceAddType& opt)
+{
+    if (!is_valid_rc_sequence(key, value))
+    {
+        if (!is_valid_rc_key(key))
+            LOG_ERROR << "Invalid key '" << key << "' or not rc configurable";
```

Comment:
I thought we were throwing errors on `std::cout`, when to use LOG_ERROR and std?

Reply:
logs are currently displayed through `std::cerr`.
we use `Console::print` for information that needs to be printed to user on `std::cout` to be able to silent it when using `--json`, `--quiet` or other similar flags

Reply:
Ah, okay, so `std::cout` is for info only and errors should be handled by `std::cerr`?

Reply:
you should never use `std::cout` directly but prefer `Console::print` to be able to silent it.
every logs messages you go to the appropriate level using macros (`LOG_INGO`, LOG_DEBUG`, etc.). For the moment yes those log messages go to the stderr there could be more sinks (file, etc.)